stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1168,6 +1168,48 @@ LogicalResult DotGeneralOp::reifyReturnTypeShapes(
   return success();
 }
 
+LogicalResult DotGeneralOp::inferReturnTypes(
+    MLIRContext*, Optional<Location>, ValueRange operands,
+    DictionaryAttr attributes, RegionRange regions,
+    SmallVectorImpl<Type>& inferredReturnTypes) {
+  DotGeneralOp::Adaptor adaptor(operands, attributes, regions);
```

Comment:
Hey, according to the verifiiers/type inference plan (Will share the RFC in Github later) how about we merge verify() in to this infer function? It can be done by this:
1. Remove `let hasVerifier = 1;` in `stablehlo/dialect/StablehloOps.td`;
2. Remove `DotGeneralOp::verify()` and copy all the code below this "adaptor(...)" line;
3. Change `this->dot_dimension_numbers()` -> `adaptor.dot_dimension_numbers()` and similar for others, because this is an static method and there's no `this` op.
4. Change `return emitOpError() << a << b << c...` to `return emitOptionalError(location, a, b, c...)
I think no changes need for tests or others. Please let me know any questions you may have.